lib/discovery.js

```diff
@@ -1,21 +1,46 @@
 module.exports = mixinDiscovery;
 
-function mixinDiscovery(MySQL) {
+/*!
+ * @param {MySQL} MySQL connector class
+ * @param {Object} mysql mysql driver
+ */
+function mixinDiscovery(MySQL, mysql) {
   var async = require('async');
 
   function paginateSQL(sql, orderBy, options) {
     options = options || {};
-    var limit = '';
+    var limitClause = '';
     if (options.offset || options.skip || options.limit) {
```

Comment:
Please note there is a story to refactor/clean up discovery/migration. This PR simply moves related code into their own files.